src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
+        long[] divisors = GetDivisors(candidate);
+        int sum = 0;
+        for (int d = 0; d < 1000; d++)
+        {
+            sum = sum + divisors[d];
+        }
+        if (sum == candidate)
+            retVal = true;
+        return retVal;
+    }
+
+
+    private static long[] GetDivisors(long candidate) {
+        long[] divisors = new long[];
+        int d = 0;
+        for (long i = 0; i < candidate; i++) {
+            long foo = candidate / i;
+            if (foo * i == candidate) {
+                divisors[d] = i;
+                d = d + 1;
+            }
+        }
+        return divisors;
+    }
```

Comment:
- this method name do not follow the Java code conventions, it should be `getDivisors`.
- the array `divisors` is not initiated with size. We don't know the numbers of possible divisors, so lets change that to a list and add as needed.
- the `d` variable is useless
- we can use the `%` operator to check if the number is really an divisor.
- at the end we transform back the list to an array if the method return type cannot be changed.

```suggestion
  private static Long[] getDivisors(long candidate) {
    List<Long> divisors = new ArrayList<>();
    for (long i = 1; i < candidate; i++) {
      if (candidate % i == 0) {
        divisors.add(i);
      }
    }
    return divisors.toArray(new Long[divisors.size()]);
  }
```